app/models/user.rb

```diff
@@ -2,18 +2,24 @@
 class User < ActiveRecord::Base
   has_paper_trail
 
-  has_many   :groups_users, dependent: :destroy
-  has_many   :groups, through: :groups_users
+  belongs_to :department
+
+  has_many :groups_users, dependent: :destroy
+  has_many :groups, through: :groups_users
 
   # user is creator
-  has_many   :created_rentals, class_name: Rental.name, foreign_key: :creator_id
+  has_many :created_rentals, class_name: Rental.name, foreign_key: :creator_id
   # user is renter
-  has_many   :rented_rentals, class_name: Rental.name, foreign_key: :renter_id
-
-  belongs_to :department
-
-  validates :first_name, :last_name, :spire_id, :phone, :email, :department, presence: true
-  validates :spire_id, length: { is: 8 }, uniqueness: true
+  has_many :rented_rentals, class_name: Rental.name, foreign_key: :renter_id
+
+  validates :first_name, :last_name, :spire_id,
+            :department, :phone, :email, presence: true
+  validates :spire_id, uniqueness: true,
+             format: { with: /\d{8}/, message: 'SpireId should be 8 digits' }
+  validates :phone, format: { with: /\d{10}/,
```

Comment:
I wonder if here it would be better to account for different formats of phone numbers instead of just checking for 10 digits (which also only accounts for US numbers.. I'm not sure if we'd ever get an international number on this app but currently the app wouldn't allow it).
What about something like
`validates :phone, format: {
      :with => /\(?[0-9]{3}\)?-[0-9]{3}-[0-9]{4}/,
      :message => "- Phone numbers must be in xxx-xxx-xxxx format."
}` which looks for xxx-xxx-xxxx format but also allows optional parenthesis. (Again still doesn't account for non-US numbers)

Reply:
I agree with that; we should allow for international numbers. It's an edge case, but we are a fairly international campus here.
I think validating that the phone number is a digit before saving is sufficient.
I would also allow entering the phone number in any way the person sees fit and then stripping any non-numeric characters before saving. For example, we should allow someone to type in `413-545-0056` or `1+413-545-0056` or `(413)545-0056`, which are all a) valid ways to display or type a phone number and b) allowed on practically any other website (which isn't shite) that asks for a number. Then use a `before_save` block to strip any non alphanumeric characters.
If a person enters in a letter, perhaps we should send an error, because it probably means they sausage-fingered and should check the number.